### PR TITLE
docs: add contributing guide

### PR DESCRIPTION
**Pull request by devadvocado**
### Summary

Adding a contributors guide to Kuma repository with best practice guidelines for folks who want to contribute.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+Your change will be included in the subsequent release Changelog, and we will
+not forget to include your name if you are an external contributor. :wink:
+
+[Back to TOC](#table-of-contents)
+
+#### Git branches
+
+We work on two branches: `master`, where non-breaking changes land, and 
+`feature/latest`, where important features or breaking changes land in-between
+major releases.
```
Comment (jakubdyszkiewicz) [inline CONTRIBUTING.md]: How does it work? When `feature/latest` is rebased to `master? Who is responsible for it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+When contributing to Kuma, this distinction is important. Please ensure that
+you are basing your work on top of the appropriate branch, it might save you
+some time down the road!
+
+If you have write access to the GitHub repository, please follow the following
+naming scheme when pushing your branch(es):
+
+- `feat/foo-bar` for new features
```
Comment (jakubdyszkiewicz) [inline CONTRIBUTING.md]: Why this is `feat/` but branch for breaking changes is `feature/`. Is this intentional?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+  a test, adding a test, improving its reliability, etc...)
+- **docs**: Changes to the README.md, this file, or other such documents
+- **style**: Changes that do not affect the meaning of the code (white-space
+  trimming, formatting, etc...)
+- **perf**: A code change that significantly improves performance
+- **refactor**: A code change that neither fixes a bug nor adds a feature, and
+  is too big to be considered just `perf`
+- **chore**: Maintenance changes related to code cleaning that isn't
+  considered part of a refactor, build process updates, dependency bumps, or
+  auxiliary tools and libraries updates (LuaRocks, Travis-ci, etc...).
```
Comment (jakubdyszkiewicz) [inline CONTRIBUTING.md]: We neither use LuaRocks nor Travis-ci here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+patch: will they understand your changes and reasoning just by reading your
+commit history? Will they find unrelated changes in a particular commit? They
+shouldn't!
+
+Writing meaningful commit messages that follow our commit message format will
+also help you respect this mantra (see the below section).
+
+[Back to TOC](#table-of-contents)
+
+#### Commit message format
```
Comment (jakubdyszkiewicz) [inline CONTRIBUTING.md]: I'm all in for the one format, but we used different one. It was `scope: msg` like `control-plane: remove http port`. Similar like in Are we all in for this change?

What about merging? For now we've used merge commits. Should we start doing rebase & merge? 
@yskopets @gschier @subnetmarco

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+- The tense of your message must be **present**
+- Your message must be prefixed by a type, and a scope
+- The header of your message should not be longer than 50 characters
+- A blank line should be included between the header and the body
+- The body of your message should not contain lines longer than 72 characters
+
+Here is a template of what your commit message should look like:
+
+```
+<type>(<scope>) <subject>
```
Comment (jakubdyszkiewicz) [inline CONTRIBUTING.md]: I like the defined types. Can we have a list of defined scopes?

File: CONTRIBUTING.md (hunk 1 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: It was something @gszr mentioned when I asked if we had a `next` branch like in Kong repo. It seems like folks mostly just merge to master right now so let's just take this out until we have a more clear cut workflow.

File: CONTRIBUTING.md (hunk 2 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: feature branch in docs removed, no longer relevant

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+All of the following tasks are noble and worthy contributions that you can
+make without coding:
+
+- Reporting a bug (see the [report bugs](#where-to-report-bugs) section)
+- Helping other members of the community on Slack
+- Fixing a typo in the code
+- Fixing a typo in the documentation at https://kuma.io/docs (see
+  the [documentation contribution](#improving-the-documentation) section)
+- Providing your feedback on the proposed features and designs
+- Reviewing Pull Requests
```
Comment (kbarnard10) [inline CONTRIBUTING.md]: Add: Writing a blog post on how you're using Kuma

File: CONTRIBUTING.md (hunk 3 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: artifacts of the kong doc, removed

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+The repository is also hosted on GitHub at:
+https://github.com/Kong/kuma-website
+
+[Back to TOC](#table-of-contents)
+
+### Submitting a patch
+
+Feel free to contribute fixes or minor features, we love to receive Pull
+Requests! If you are planning to develop a larger feature, come talk to us
+first!
```
Comment (kbarnard10) [inline CONTRIBUTING.md]: Is there a place they can come talk to us? Email or Slack?

File: CONTRIBUTING.md (hunk 5 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: added following scopes:
- kuma-cp
- kuma-dp
- injector
- kumactp
- deps
- conf
- *

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+      - [Commit message format](#commit-message-format)
+        - [Type](#type)
+        - [Subject](#subject)
+        - [Body](#body)
+        - [Footer](#footer)
+        - [Examples](#examples)
+      - [Writing tests](#writing-tests)
+    - [Contributor T-shirt](#contributor-t-shirt)
+
+## Where to seek for help?
```
[the remark this answers is not in the document]
Comment (devadvocado) [reply CONTRIBUTING.md]: Slack info below. Any more channels you want to add?

File: CONTRIBUTING.md (hunk 7 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: added a link to the slack section above

File: CONTRIBUTING.md (hunk 6 above)
Comment (devadvocado) [reply CONTRIBUTING.md]: added 👍

File: CONTRIBUTING.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (kbarnard10) [reply CONTRIBUTING.md]: Meant to say change: Where to seek for help? -> Where to seek help?

[the remark this answers is not in the document]
Comment (devadvocado) [reply CONTRIBUTING.md]: 🤦‍♂ my bad, updated it!

File: CONTRIBUTING.md (hunk 4 above)
Comment (gszr) [reply CONTRIBUTING.md]: I'd argue in favor of changing to the format using in Kong; having both "type" and "scope" is very helpful for code archaeology. Plus, that format is already used in the organization.

On merging, I usually do rebase and merge; however, there are cases where a merge commit can be helpful -- same for squash and merge.

**Pull request merged**